velox/functions/prestosql/ArrayFunctions.h

```diff
@@ -697,4 +697,76 @@ struct ArrayUnionFunction {
   }
 };
 
+/// This class implements the array_remove function.
+///
+/// DEFINITION:
+/// array_remove(x, element) -> array
+/// Remove all elements that equal element from array x.
+template <typename T>
+struct ArrayRemoveFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  // Fast path for primitives.
+  template <typename Out, typename In, typename E>
+  void call(Out& out, const In& inputArray, E element) {
+    for (const auto& item : inputArray) {
+      if (item.has_value()) {
+        if (element != item.value()) {
+          auto& newItem = out.add_item();
+          newItem = item.value();
+        }
+      } else {
+        out.add_null();
+      }
+    }
+  }
+
+  // Generic implementation.
+  void call(
+      out_type<Array<Generic<T1>>>& out,
+      const arg_type<Array<Generic<T1>>>& array,
+      const arg_type<Generic<T1>>& element) {
+    static constexpr CompareFlags kFlags = {
+        false, false, /*euqalsOnly*/ true, true /*stopAtNull*/};
+    for (const auto& item : array) {
+      if (item.has_value()) {
+        auto result = element.compare(item.value(), kFlags);
+        VELOX_USER_CHECK(
```

Comment:
why do we not have this check for primitives?

Reply:
primitives cannot be arrays/maps right?